### ISSUE TITLE
val_loss and mAP plot for every iteration

### ISSUE THREAD

**Issue by xQsM3** [facebookresearch/detectron2#2448]
I would like to plot val_loss, train_loss and mAP for training. Therefore, I used a custom hook in a child class of the DefaultTrainer. Now, when I set e.g. cfg.TEST.EVAL_PERIOD = 500, the mAP and the val_loss is calculated every 500 iterations with the full validation set. 

But I would like to calculate the val_loss and the mAP every iteration but instead from the FULL validation set, just from a few images of the validation set (e.g. equal to bach size).  I checked the config file of the Default Trainer:
https://github.com/facebookresearch/Detectron/blob/master/detectron/core/config.py

But i did not find anything like that. Is there any chance to do this?

Thank you for your help

**Comment by ppwwyyxx** [comment]
There is no such feature.

They can be implemented similar to how evaluation on full dataset is implemented. Related tutorial: https://detectron2.readthedocs.io/tutorials/training.html
